proto/sync/sync.proto

```diff
@@ -32,10 +43,92 @@ message ChangeProofRequest {
   uint32 bytes_limit = 6;
 }
 
-// KeyValue represents a single key and its value.
-message KeyValue {
+message GetChangeProofResponse {
+  ChangeProof proof = 1;
+}
+
+message VerifyChangeProofRequest {
+  ChangeProof proof = 1;
+}
+
+message VerifyChangeProofResponse {
+  rpcdb.Error error = 1;
+}
+
+message CommitChangeProofRequest {
+  ChangeProof proof = 1;
+}
+
+message CommitChangeProofResponse {
+  rpcdb.Error error = 1;
+}
+
+message GetProofRequest {
   bytes key = 1;
-  bytes value = 2;
+}
+
+message GetProofResponse {
+  Proof proof = 1;
+}
+
+// A GetRangeProofRequest requests the key-value pairs in a given key range
+// at a specific revision.
```

Comment:
Honestly idk if we really need comments on the proto stuff since it's documented on the `sync.SyncableDB` interface...these comments were carried over from the branch this is based on.